internal/redfishwrapper/virtual_media.go

```diff
@@ -0,0 +1,74 @@
+package redfishwrapper
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	rf "github.com/stmcginnis/gofish/redfish"
+)
+
+// Set the boot device for the system.
+func (c *Client) SetVirtualMedia(ctx context.Context, kind string, mediaUrl string) (ok bool, err error) {
+	managers, err := c.Managers(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	var mediaKind rf.VirtualMediaType
+	switch kind {
+	case "CD":
+		mediaKind = rf.CDMediaType
+	case "Floppy":
+		mediaKind = rf.FloppyMediaType
+	case "USBStick":
+		mediaKind = rf.USBStickMediaType
+	case "DVD":
+		mediaKind = rf.DVDMediaType
+	default:
+		return false, errors.New("invalid media type")
+	}
+
+	for _, manager := range managers {
+		virtualMedia, err := manager.VirtualMedia()
+		if err != nil {
+			return false, err
+		}
+		for _, media := range virtualMedia {
+			if media.Inserted {
+				err = media.EjectMedia()
```

Comment:
This loop seems to eject all existing virtualMedia, is this the expected behaviour - should this only eject the mediaKind its attempting to set?

Reply:
Happy to change it. I wasn't 100% sure whether BMCs allow multiple kinds of virtual media to be attached at once, and this approach guarantees there won't be.

Reply:
Sure, we can go with this approach for now.